src/vaadin-grid-scroller.html

```diff
@@ -280,6 +280,27 @@
       _scrollHandler() {
         const delta = this.$.table.scrollTop - this._scrollPosition;
         this._accessIronListAPI(super._scrollHandler);
+
+        if (this._physicalCount !== 0) {
+          const isScrollingDown = delta >= 0;
+          const reusables = this._getReusables(!isScrollingDown);
+
+          if (reusables.indexes.length) {
+            // After running super._scrollHandler, fix internal properties to workaround an iron-list issue.
+            // See https://github.com/vaadin/web-components/issues/1691
+            this._physicalTop = reusables.physicalTop;
+
+            if (isScrollingDown) {
+              this._virtualStart -= reusables.indexes.length;
+              this._physicalStart -= reusables.indexes.length;
+            } else {
+              this._virtualStart += reusables.indexes.length;
+              this._physicalStart += reusables.indexes.length;
+            }
+            this._update(reusables.indexes, !isScrollingDown);
+          }
```

Comment:
This is basically a copy of the [logic in iron-list's `_scrollHandler`](https://github.com/vaadin/vaadin-grid/blob/18018f09c0aeb9a4e1fa535d4b35338f6e52c4a5/src/iron-list.html#L379-L388). In certain situation (with large enough item count and scrollbar scrolling) the logic may not get executed when it should.